course/views.py

```diff
@@ -106,8 +106,23 @@ def get(self, request, *args, **kwargs):
         elif lti_msg:
             messages.info(request, html.escape(lti_msg))
 
-        return super().get(request, *args, **kwargs)
-
+        later_instance = None
+
+        if self.instance.is_past:
+            course = self.instance.course
+            now = timezone.now()
+            try:
+                later_instance = CourseInstance.objects.get_visible(request.user).filter(
+                    ending_time__gte=now,
+                    course=course
+                ).latest('starting_time')
+            except ObjectDoesNotExist:
+                pass
+
+        return super().get(request,
+                            *args,
+                            **kwargs,
+                            later_instance=later_instance)
```

Comment:
If the `later_instance` exists in the `kwargs`, then this line will crash. It might be a good idea to always overwrite the field using line `kwargs['later_instance'] = later_instance`.